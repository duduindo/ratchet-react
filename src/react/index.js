import Header from './Header';

export {
  Header as RTHeader
};

