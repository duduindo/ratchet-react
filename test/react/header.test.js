import React from 'react';
import {shallow, render, mount} from 'enzyme';


//  My scripts
import { RTHeader } from '../../src/react/';
//import { RTHeader } from '../../lib/common.min.js';


describe('Header', () => {

  it('Snapshot with Children', () => {
    const component = shallow(
      <RTHeader className="header-custom">
        <h1>Title</h1>
        <ul>
          <li>1</li>
          <li>2</li>
          <li>3</li>
        </ul>
      </RTHeader>
    );

    expect(component).toMatchSnapshot();
  });

  it('Selector', () => {
    const component = shallow(
      <RTHeader className="header-custom">
        <h1>Title</h1>
      </RTHeader>
    );

    expect(component.find('.bar.bar-nav.header-custom').length).toEqual(1);
  });

})
