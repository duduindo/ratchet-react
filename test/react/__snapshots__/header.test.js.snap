// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot with Children 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <header
    className="bar bar-nav header-custom"
>
    <h1>
        Title
    </h1>
    <ul>
        <li>
            1
        </li>
        <li>
            2
        </li>
        <li>
            3
        </li>
    </ul>
</header>,
  "nodes": Array [
    <header
      className="bar bar-nav header-custom"
>
      <h1>
            Title
      </h1>
      <ul>
            <li>
                  1
            </li>
            <li>
                  2
            </li>
            <li>
                  3
            </li>
      </ul>
</header>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Header
        className="header-custom"
>
        <h1>
                Title
        </h1>
        <ul>
                <li>
                        1
                </li>
                <li>
                        2
                </li>
                <li>
                        3
                </li>
        </ul>
</Header>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <h1>
              Title
</h1>,
            <ul>
              <li>
                            1
              </li>
              <li>
                            2
              </li>
              <li>
                            3
              </li>
</ul>,
          ],
          "className": "header-custom",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <header
          className="bar bar-nav header-custom"
>
          <h1>
                    Title
          </h1>
          <ul>
                    <li>
                              1
                    </li>
                    <li>
                              2
                    </li>
                    <li>
                              3
                    </li>
          </ul>
</header>,
        "_debugID": 2,
        "_renderedOutput": <header
          className="bar bar-nav header-custom"
>
          <h1>
                    Title
          </h1>
          <ul>
                    <li>
                              1
                    </li>
                    <li>
                              2
                    </li>
                    <li>
                              3
                    </li>
          </ul>
</header>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header
    className="header-custom"
>
    <h1>
        Title
    </h1>
    <ul>
        <li>
            1
        </li>
        <li>
            2
        </li>
        <li>
            3
        </li>
    </ul>
</Header>,
}
`;
